core/src/main/scala/org/bitcoins/core/crypto/AesCrypt.scala

```diff
@@ -63,118 +306,77 @@ object AesCrypt {
     */
   def decrypt(
       encrypted: AesEncryptedData,
-      password: AesPassword): Either[AesDecryptionException, ByteVector] = {
-    if (password.value.isEmpty) {
-      Left(AesException.EmptyPasswordException)
-    } else {
-      val secret = getSecretKey(password, encrypted.salt)
-      val cipher = decryptionCipher(secret, encrypted.iv)
-
-      val decryptionAttempt = Try {
-        val plainText = cipher.doFinal(encrypted.cipherText.toArray)
-        ByteVector(plainText)
-      }
-
-      decryptionAttempt match {
-        case Success(bytes) => Right(bytes)
-        // here we assume that bad padding is because of a bad password
-        // provided. assuming that our implementation is correct, correct
-        // passwords should never to lead to bad padding
-        case Failure(_: BadPaddingException) =>
-          Left(AesException.BadPasswordException)
-        case Failure(exception) => throw exception
-      }
+      key: AesKey): Either[AesDecryptionException, ByteVector] = {
+    val cipher = decryptionCipher(key, encrypted.iv)
+
+    val decryptionAttempt = Try {
+      val plainText = cipher.doFinal(encrypted.cipherText.toArray)
+      ByteVector(plainText)
+    }
+
+    decryptionAttempt match {
+      case Success(bytes) => Right(bytes)
+      // here we assume that bad padding is because of a bad password
+      // provided. assuming that our implementation is correct, correct
+      // passwords should never to lead to bad padding
```

Comment:
is there a way we can test this with property based testing?

Reply:
I'm not sure how. Generally speaking, I believe its impossible to know if decryption failed because of bad data or a bad key.